subsys/bluetooth/host/gatt.c

```diff
@@ -1383,29 +1383,38 @@ static void gatt_indicate_rsp(struct bt_conn *conn, u8_t err,
 }
 
 static int gatt_send(struct bt_conn *conn, struct net_buf *buf,
-		     bt_att_func_t func, void *params,
+		     bt_att_func_t func, void *params, size_t params_size,
 		     bt_att_destroy_t destroy)
 {
-	int err;
+	int ret;
 
 	if (params) {
-		struct bt_att_req *req = params;
+		struct bt_att_req *req;
+
+		/* Allocate new request */
+		req = bt_att_req_alloc(params_size, K_FOREVER);
+		memcpy(req->params, params, params_size);
 
 		req->buf = buf;
 		req->func = func;
 		req->destroy = destroy;
 
-		err = bt_att_req_send(conn, req);
+		ret = bt_att_req_send(conn, req);
+		if (ret) {
+			bt_att_req_free(req);
+		} else {
+			ret = POINTER_TO_INT(req);
```

Comment:
This would be problematic for 64-bit system, (AFAIK we have some support for that now). Is there perhaps some form of ID that can be used to reference mem blocks? net_buf has something, but I couldn't find something for k_mem

Reply:
Not sure why would it be a problem on 64 bit, afaik int would be 64 bits as well.

Reply:
No, int would be 32 bit. POINTER_TO_INT casts to uintptr_t which is defined as long unsigned int